### PR TITLE
short circuit docker image creation if image with same tag already ex…

### PR DESCRIPTION
**Pull request by danabens**
If the docker image for integration tests has already been created (with the same tag) don't attempt to recreate it.

```
tests/integ/test_experiment.py .......                                                                                                                                                             [  7%]
tests/integ/test_tracker.py .......                                                                                                                                                                [ 14%]
tests/integ/test_trial.py .....                                                                                                                                                                    [ 20%]
tests/integ/test_trial_component.py ....                                                                                                                                                           [ 24%]
tests/integ-jobs/test_track_from_processing_job.py s                                                                                                                                               [ 25%]
tests/integ-jobs/test_track_from_training_job.py s                                                                                                                                                 [ 26%]
tests/unit/test_base_types.py ...........                                                                                                                                                          [ 37%]
tests/unit/test_environment.py .....                                                                                                                                                               [ 43%]
tests/unit/test_experiment.py ............                                                                                                                                                         [ 55%]
tests/unit/test_metrics.py .........                                                                                                                                                               [ 65%]
tests/unit/test_tracker.py ...................                                                                                                                                                     [ 85%]
tests/unit/test_trial.py ........                                                                                                                                                                  [ 93%]
tests/unit/test_trial_component.py ......                                                                                                                                                          [100%]

================================================================================= 93 passed, 2 skipped in 41.33 seconds ==================================================================================
py36 run-test: commands[1] | coverage report --fail-under=80
Name                                                                     Stmts   Miss  Cover
--------------------------------------------------------------------------------------------
.tox/py36/lib/python3.6/site-packages/smexperiments/__init__.py              0      0   100%
.tox/py36/lib/python3.6/site-packages/smexperiments/_base_types.py          69      4    94%
.tox/py36/lib/python3.6/site-packages/smexperiments/_boto_functions.py      34      0   100%
.tox/py36/lib/python3.6/site-packages/smexperiments/_environment.py         36      1    97%
.tox/py36/lib/python3.6/site-packages/smexperiments/_utils.py               36     12    67%
.tox/py36/lib/python3.6/site-packages/smexperiments/api_types.py            91      5    95%
.tox/py36/lib/python3.6/site-packages/smexperiments/experiment.py           30      0   100%
.tox/py36/lib/python3.6/site-packages/smexperiments/metrics.py              76      4    95%
.tox/py36/lib/python3.6/site-packages/smexperiments/tracker.py             110      8    93%
.tox/py36/lib/python3.6/site-packages/smexperiments/trial.py                45      4    91%
.tox/py36/lib/python3.6/site-packages/smexperiments/trial_component.py      41      1    98%
--------------------------------------------------------------------------------------------
TOTAL                                                                      568     39    93%
________________________________________________________________________________________________ summary _________________________________________________________________________________________________
  py36: commands succeeded
  congratulations :)
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/integ-jobs/test_track_from_processing_job.py [hunk 1]
```diff
@@ -33,7 +35,7 @@ def wait_for_job(job, sagemaker_client):
                 sys.stdout.flush()
                 time.sleep(30)
 
-
+#@pytest.mark.skip(reason="temporary skipping for expediency")
```
Comment (l2yao) [inline tests/integ-jobs/test_track_from_processing_job.py]: Can we revert this if this is no longer being used?

File: tests/integ-jobs/test_track_from_training_job.py [hunk 2]
```diff
@@ -59,7 +60,7 @@ def wait_for_job(job, sagemaker_client):
                 sys.stdout.flush()
                 time.sleep(30)
 
-
+#@pytest.mark.skip(reason="temporary skipping for expediency")
```
Comment (l2yao) [inline tests/integ-jobs/test_track_from_training_job.py]: can we revert if this is no longer used?

File: tests/conftest.py [hunk 3]
```diff
@@ -308,13 +304,24 @@ def docker_image(boto_model_file):
         else:
             raise
 
-    tag = '{}/{}:{}'.format(registry, repository_name, '1.0.0')[8:]
-
     # pull existing image for layer cache
     try:
         client.images.pull(tag, auth_config={'username': username, 'password': password})
+        # the image with this tag already exists
```
Comment (l2yao) [inline tests/conftest.py]: should we get the image from pull above, and actually check if this image exists? and then return the tag?

Comment (danabens) [reply tests/conftest.py]: the tag is part of the pull request, if there isn't a not found exception we can safely assume the image exists with the tag we provided

File: tests/integ-jobs/test_track_from_training_job.py (hunk 2 above)
Comment (danabens) [reply tests/integ-jobs/test_track_from_training_job.py]: removed

File: tests/integ-jobs/test_track_from_processing_job.py (hunk 1 above)
Comment (danabens) [reply tests/integ-jobs/test_track_from_processing_job.py]: removed

File: tests/conftest.py (hunk 3 above)
Comment (l2yao) [reply tests/conftest.py]: Sounds good to me

**Pull request merged**